image/manifest.go

```diff
@@ -59,54 +328,65 @@ func ReadSeedManifest(manifestFile string) (map[string]snap.Revision, error) {
 		}
 
 		tokens := strings.Fields(line)
-		// Expect exactly two tokens
-		if len(tokens) != 2 {
-			return nil, fmt.Errorf("line is illegally formatted: %q", line)
-		}
-
-		snapName := tokens[0]
-		revString := tokens[1]
-		if err := snap.ValidateName(snapName); err != nil {
-			return nil, err
-		}
 
-		rev, err := snap.ParseRevision(revString)
-		if err != nil {
-			return nil, err
+		if len(tokens) == 1 && strings.Contains(tokens[0], "/") {
+			// Pinned validation-set: <account-id>/<name>=<sequence>
+			if err := parsePinnedValidationSet(sm, tokens[0]); err != nil {
+				return nil, err
+			}
+		} else if len(tokens) == 2 {
+			if strings.Contains(tokens[0], "/") {
+				// Unpinned validation-set: <account-id>/<name> <sequence>
+				if err := parseUnpinnedValidationSet(sm, tokens[0], tokens[1]); err != nil {
+					return nil, err
+				}
+			} else {
+				// Snap revision: <snap> <revision>
+				if err := parseSnapRevision(sm, tokens[0], tokens[1]); err != nil {
+					return nil, err
+				}
+			}
+		} else {
+			return nil, fmt.Errorf("line is illegally formatted: %q", line)
```

Comment:
(super nitpick) maybe more consistent as `cannot parse line %q` or `cannot parse illegally formated line %q`?

Reply:
I like the first one! Fixed it